src/main.cpp

```diff
@@ -36,18 +39,74 @@ std::optional<float> Sensor::get_dist() const {
 	auto const obj = get_obj();
 	if (obj.has_value()) {
 		int const y_height = obj.value().y_middle_coord;
-		return measures::GOAL_HEIGHT - measures::SENSOR_HEIGHT / std::tan(measures::LENS_ANGLE - std::atan((1 - y_height / 200) * std::tan(measures::VERTICAL_FOV / 2)));
+		return dimensions::GOAL_HEIGHT - dimensions::SENSOR_HEIGHT / std::tan(dimensions::LENS_ANGLE - std::atan((1 - y_height / 200) * std::tan(constants::SENSOR_VERTICAL_FOV / 2)));
 	}
 	return {};
 }
 
+Flywheel::Flywheel(uint8_t const left_port, uint8_t const right_port, uint8_t const piston_port) : m_left_motor{left_port}, m_right_motor{right_port}, m_piston{piston_port} {}
+
+void Flywheel::spin_up() {
+	m_left_motor.move_velocity(constants::FLYWHEEL_ON_VELOCITY);
+	m_right_motor.move_velocity(constants::FLYWHEEL_ON_VELOCITY);
+	m_on = true;
+}
+
+void Flywheel::disengage() {
+	m_left_motor.move_voltage(0);
+	m_right_motor.move_voltage(0);
+	m_on = false;
+}
+
+void Flywheel::aim(float const distance) const {
+	float const secant = 1/std::cos(dimensions::LAUNCH_ANGLE);
+	int16_t velocity = std::sqrt((-1 * std::pow(distance, 2) * constants::GRAVITY * std::pow(secant, 2))/(2 * (dimensions::GOAL_HEIGHT - (distance * secant * std::sin(dimensions::LAUNCH_ANGLE))) - dimensions::LAUNCH_HEIGHT)) * constants::FLYWHEEL_PROPORTION;
```

Comment:
I would prefer if this were extracted into some intermediate variables, to make it clear how the items are interacting.